# Data-Driven Insights for Northwind Traders

## Introduction

This project focuses on the Northwind database, which provides a real-world-like platform for exploring sales data. A schema of the database can be found [here](https://s3.amazonaws.com/dq-content/777/1.2-m777.svg).

Northwind Traders is a hypothetical, international gourmet food distributor, and the focuses of the project are as follows:

- Evaluating employee performance to boost productivity
- Understanding product sales and category performance to optimise inventory and marketing strategies
- Analysing sales growth to identify trends, monitor company progress, and make more accurate forecasts
- Evaluating customer purchase behaviour to target high-value customers with promotional incentives

Skills Demonstrated:
- Window Functions
- CTEs
- Data-informed Decisions 

## Getting to Know the Data

Obtaining a list of all tables and views in the PostgreSQL database.

In [2]:
%%sql
SELECT table_name AS name,
       table_type AS type
  FROM information_schema.tables
 WHERE table_schema = 'public' AND table_type IN ('BASE TABLE', 'VIEW');

 * postgresql://postgres:***@localhost:5432/northwind
17 rows affected.


[('territories', 'BASE TABLE'),
 ('order_product_details_view', 'VIEW'),
 ('order_details', 'BASE TABLE'),
 ('employee_territories', 'BASE TABLE'),
 ('us_states', 'BASE TABLE'),
 ('customers', 'BASE TABLE'),
 ('orders', 'BASE TABLE'),
 ('employees', 'BASE TABLE'),
 ('shippers', 'BASE TABLE'),
 ('products', 'BASE TABLE'),
 ('categories', 'BASE TABLE'),
 ('suppliers', 'BASE TABLE'),
 ('region', 'BASE TABLE'),
 ('customer_demographics', 'BASE TABLE'),
 ('customer_customer_demo', 'BASE TABLE'),
 ('order_details_view', 'VIEW'),
 ('employee_order_view', 'VIEW')]

This query retrieves information about all tables and views in the database schema 'public' and displays their names and types.

## Exploring Tables

### Customers Table

In [3]:
%%sql 

SELECT *
  FROM customers
 LIMIT 2;

 * postgresql://postgres:***@localhost:5432/northwind
2 rows affected.


[('ALFKI', 'Alfreds Futterkiste', 'Maria Anders', 'Sales Representative', 'Obere Str. 57', 'Berlin', None, '12209', 'Germany', '030-0074321', '030-0076545'),
 ('ANATR', 'Ana Trujillo Emparedados y helados', 'Ana Trujillo', 'Owner', 'Avda. de la Constitución 2222', 'México D.F.', None, '05021', 'Mexico', '(5) 555-4729', '(5) 555-3745')]

The customers table contains key columns including customer_id, company_name, contact_name, contact_title, address, city, region, postal_code, country, phone, and fax. The data represents customers from different countries.

### Orders Table

In [4]:
%%sql 

SELECT *
  FROM orders
 LIMIT 2;

 * postgresql://postgres:***@localhost:5432/northwind
2 rows affected.


[(10248, 'VINET', 5, datetime.date(1996, 7, 4), datetime.date(1996, 8, 1), datetime.date(1996, 7, 16), 3, 32.38, 'Vins et alcools Chevalier', "59 rue de l'Abbaye", 'Reims', None, '51100', 'France'),
 (10249, 'TOMSP', 6, datetime.date(1996, 7, 5), datetime.date(1996, 8, 16), datetime.date(1996, 7, 10), 1, 11.61, 'Toms Spezialitäten', 'Luisenstr. 48', 'Münster', None, '44087', 'Germany')]

Each row corresponds to a specific order placed by a customer, including details such as shipping information, order dates, and freight costs.

### Products Table

In [5]:
%%sql 

SELECT *
  FROM products
 LIMIT 2;

 * postgresql://postgres:***@localhost:5432/northwind
2 rows affected.


[(1, 'Chai', 8, 1, '10 boxes x 30 bags', 18.0, 39, 0, 10, 1),
 (2, 'Chang', 1, 1, '24 - 12 oz bottles', 19.0, 17, 40, 25, 1)]

Each row corresponds to a specific product, including details such as supplier information, pricing, stock levels, reorder information, and whether the product is discontinued.

### Order Details Table

In [6]:
%%sql 

SELECT *
  FROM order_details
 LIMIT 2;

 * postgresql://postgres:***@localhost:5432/northwind
2 rows affected.


[(10248, 11, 14.0, 12, 0.0), (10248, 42, 9.8, 10, 0.0)]

Each row corresponds to a specific product ordered within an order, including details such as unit price, quantity ordered, and any applicable discounts for the order.

## Creating Views

### Combining orders and customers tables

This first query will provide more detailed information about each order by combining data from the orders and customers tables.

In [7]:
%%sql

CREATE VIEW order_details_view AS
SELECT o.order_id,
       c.company_name,
       c.contact_name,
       o.order_date
  FROM orders AS o
  JOIN customers AS c ON o.customer_id = c.customer_id;

 * postgresql://postgres:***@localhost:5432/northwind
(psycopg2.errors.DuplicateTable) relation "order_details_view" already exists

[SQL: CREATE VIEW order_details_view AS
SELECT o.order_id,
       c.company_name,
       c.contact_name,
       o.order_date
  FROM orders AS o
  JOIN customers AS c ON o.customer_id = c.customer_id;]
(Background on this error at: https://sqlalche.me/e/20/f405)


In [8]:
%%sql

SELECT *
  FROM order_details_view 
 LIMIT 5;

 * postgresql://postgres:***@localhost:5432/northwind
5 rows affected.


[(10248, 'Vins et alcools Chevalier', 'Paul Henriot', datetime.date(1996, 7, 4)),
 (10249, 'Toms Spezialitäten', 'Karin Josephs', datetime.date(1996, 7, 5)),
 (10250, 'Hanari Carnes', 'Mario Pontes', datetime.date(1996, 7, 8)),
 (10251, 'Victuailles en stock', 'Mary Saveley', datetime.date(1996, 7, 8)),
 (10252, 'Suprêmes délices', 'Pascale Cartrain', datetime.date(1996, 7, 9))]

### Combining order_details, products, and orders tables

This query will provide detailed order information including product name and quantity by combining data from the order_details, products, and orders tables.

In [9]:
%%sql 

CREATE VIEW order_product_details_view AS
SELECT o.order_id,
       p.product_name,
       od.quantity,
       o.order_date
  FROM order_details AS od
  JOIN products AS p ON od.product_id = p.product_id
  JOIN orders AS o ON od.order_id = o.order_id;

 * postgresql://postgres:***@localhost:5432/northwind
(psycopg2.errors.DuplicateTable) relation "order_product_details_view" already exists

[SQL: CREATE VIEW order_product_details_view AS
SELECT o.order_id,
       p.product_name,
       od.quantity,
       o.order_date
  FROM order_details AS od
  JOIN products AS p ON od.product_id = p.product_id
  JOIN orders AS o ON od.order_id = o.order_id;]
(Background on this error at: https://sqlalche.me/e/20/f405)


In [10]:
%%sql

SELECT *
  FROM order_product_details_view
 LIMIT 5;

 * postgresql://postgres:***@localhost:5432/northwind
5 rows affected.


[(10248, 'Queso Cabrales', 12, datetime.date(1996, 7, 4)),
 (10248, 'Singaporean Hokkien Fried Mee', 10, datetime.date(1996, 7, 4)),
 (10248, 'Mozzarella di Giovanni', 5, datetime.date(1996, 7, 4)),
 (10249, 'Tofu', 9, datetime.date(1996, 7, 5)),
 (10249, 'Manjimup Dried Apples', 40, datetime.date(1996, 7, 5))]

### Combining employees and orders tables

This query will show which employee is responsible for each order by combining data from the employees and orders tables.

In [11]:
%%sql

CREATE VIEW employee_order_view AS
SELECT e.first_name || ' ' || e.last_name as employee_name,
       o.order_id,
       o.order_date
  FROM employees AS e 
  JOIN orders AS o ON e.employee_id = o.employee_id;

 * postgresql://postgres:***@localhost:5432/northwind
(psycopg2.errors.DuplicateTable) relation "employee_order_view" already exists

[SQL: CREATE VIEW employee_order_view AS
SELECT e.first_name || ' ' || e.last_name as employee_name,
       o.order_id,
       o.order_date
  FROM employees AS e 
  JOIN orders AS o ON e.employee_id = o.employee_id;]
(Background on this error at: https://sqlalche.me/e/20/f405)


In [12]:
%%sql

SELECT *
  FROM employee_order_view
 LIMIT 5;

 * postgresql://postgres:***@localhost:5432/northwind
5 rows affected.


[('Steven Buchanan', 10248, datetime.date(1996, 7, 4)),
 ('Michael Suyama', 10249, datetime.date(1996, 7, 5)),
 ('Margaret Peacock', 10250, datetime.date(1996, 7, 8)),
 ('Janet Leverling', 10251, datetime.date(1996, 7, 8)),
 ('Margaret Peacock', 10252, datetime.date(1996, 7, 9))]

## Analysis and Insights

### Ranking Employees Sales Performance

I will now conduct a comprehensive review of the company's sales performance from an employees perspective. My objective is two-fold:

1. Identify and recognise top-performing employees to foster a culture of excellence within the organisation.
2. Identify employees who may need additional support or training to improve their sales performance.

To achieve these goals, I will rank employees based on their total sales amount. This ranking will help highlight both high-performing individuals who deserve recognition and employees who may benefit from further development opportunities.

In [13]:
%%sql

-- CTE to calculate total sales for each employee

WITH employee_total_sales AS (
    SELECT e.employee_id, e.first_name, e.last_name, 
           SUM(od.unit_price * od.quantity * (1 - od.discount)) AS "Total Sales"   
    FROM orders AS o
    JOIN order_details AS od ON o.order_id = od.order_id     
    JOIN employees AS e ON e.employee_id = o.employee_id
    GROUP BY e.employee_id
)

-- Query to retrieve employee ID, first name, last name, and their sales rank based on total sales

SELECT employee_id, first_name, last_name,
       RANK() OVER(ORDER BY "Total Sales" DESC) AS "Sales Rank"
FROM employee_total_sales;


 * postgresql://postgres:***@localhost:5432/northwind
9 rows affected.


[(4, 'Margaret', 'Peacock', 1),
 (3, 'Janet', 'Leverling', 2),
 (1, 'Nancy', 'Davolio', 3),
 (2, 'Andrew', 'Fuller', 4),
 (8, 'Laura', 'Callahan', 5),
 (7, 'Robert', 'King', 6),
 (9, 'Anne', 'Dodsworth', 7),
 (6, 'Michael', 'Suyama', 8),
 (5, 'Steven', 'Buchanan', 9)]

The top-performing employees who achieved notable sales ranks and should be recognised are Margaret Peacock (1st rank), Janet Leverling (2nd rank), and Nancy Davolio (3rd rank). On the other hand, employees who may require additional support or training to enhance their sales performance include Michael Suyama (8th rank) and Steven Buchanan (9th rank), as they are positioned lower in the sales ranking. Identifying and addressing the needs of both high-performing and struggling employees aligns with fostering a culture of excellence and continuous improvement within the organisation.

### Running Total of Monthly Sales

Now I will be working on generating a running total of monthly sales. This task involves aggregating sales data at a monthly level and calculating the cumulative total sales for each month. The objective is to create a visual representation that showcases the company's sales progress over time, making it easy to track sales trends and identify patterns such as periods of high or low sales activity.

In [14]:
%%sql

-- CTE to calculate monthly sales
WITH monthly_sales AS (
    -- Selecting truncated month and calculating total sales per month
    
    SELECT DATE_TRUNC('month', o.order_date)::DATE AS "Month",
           SUM(od.unit_price * od.quantity * (1 - od.discount)) AS "Total Sales"
      FROM orders AS o
      JOIN order_details AS od ON o.order_id = od.order_id
     GROUP BY DATE_TRUNC('month', o.order_date)::DATE
)

-- Main query to calculate running total of sales over time

SELECT "Month",
       SUM("Total Sales") OVER(ORDER BY "Month") AS "Running Total"
  FROM monthly_sales
 ORDER BY "Month";


 * postgresql://postgres:***@localhost:5432/northwind
23 rows affected.


[(datetime.date(1996, 7, 1), 27861.89512966156),
 (datetime.date(1996, 8, 1), 53347.17020040483),
 (datetime.date(1996, 9, 1), 79728.57033299239),
 (datetime.date(1996, 10, 1), 117244.29527847127),
 (datetime.date(1996, 11, 1), 162844.3404896083),
 (datetime.date(1996, 12, 1), 208083.97098282274),
 (datetime.date(1997, 1, 1), 269342.0411508011),
 (datetime.date(1997, 2, 1), 307825.6761011254),
 (datetime.date(1997, 3, 1), 346372.8962108522),
 (datetime.date(1997, 4, 1), 399405.8485997937),
 (datetime.date(1997, 5, 1), 453187.13842493534),
 (datetime.date(1997, 6, 1), 489549.9407597378),
 (datetime.date(1997, 7, 1), 540570.7982783426),
 (datetime.date(1997, 8, 1), 587858.4679665978),
 (datetime.date(1997, 9, 1), 643487.7103683471),
 (datetime.date(1997, 10, 1), 710236.9361440743),
 (datetime.date(1997, 11, 1), 753770.7449116395),
 (datetime.date(1997, 12, 1), 825169.1733755233),
 (datetime.date(1998, 1, 1), 919391.2835824591),
 (datetime.date(1998, 2, 1), 1018806.5709654673),
 (datetime.date(1998, 3, 1), 1123660.7259656242),
 (datetime.date(1998, 4, 1), 1247459.4082211715),
 (datetime.date(1998, 5, 1), 1265793.0386533642)]

The running total of monthly sales illustrates a consistent and positive growth trend from July 1996 to May 1998. The cumulative sales figures show steady month-over-month increases, reflecting sustained sales performance over time. Notable spikes in growth occur at specific intervals, highlighting periods of accelerated sales activity. This analysis provides valuable insights into the company's sales trajectory and overall business performance.

### Month-Over-Month Sales Growth

I will calculate the month-over-month sales growth rate to further analyse the sales figures presented earlier. I will use window functions, specifically the LAG function, to compare each month's sales with the previous month's sales. This calculation will help us understand the percentage change in sales from one month to the next, providing insights into periods of growth or contraction. These insights would be valuable for informing business strategy and decision-making.

In [15]:
%%sql

-- This SQL script calculates the growth rate of total sales per month compared to the previous month's sales.

-- Calculating total sales per month
WITH total_sales_per_month AS (
      SELECT EXTRACT('month' FROM o.order_date) AS Month,
             EXTRACT('year' FROM o.order_date) AS Year,
             SUM(od.unit_price * od.quantity * (1 - od.discount)) AS total_sales
      FROM orders AS o
      JOIN order_details AS od ON o.order_id = od.order_id
      GROUP BY EXTRACT('month' FROM o.order_date), EXTRACT('year' FROM o.order_date)
),

-- Calculating previous month's total sales
previous_month_total_sales AS (
      SELECT Month, Year, total_sales,
             LAG(total_sales) OVER(ORDER BY Year, Month) AS previous_months_sales
      FROM total_sales_per_month
)

-- Calculating growth rate
SELECT Year, Month,
       ((total_sales - previous_months_sales) / previous_months_sales) * 100 AS growth_rate
  FROM previous_month_total_sales;


 * postgresql://postgres:***@localhost:5432/northwind
23 rows affected.


[(Decimal('1996'), Decimal('7'), None),
 (Decimal('1996'), Decimal('8'), -8.530001451294545),
 (Decimal('1996'), Decimal('9'), 3.51624637896504),
 (Decimal('1996'), Decimal('10'), 42.20520805162909),
 (Decimal('1996'), Decimal('11'), 21.54915112904513),
 (Decimal('1996'), Decimal('12'), -0.7903823696967553),
 (Decimal('1997'), Decimal('1'), 35.40798079057388),
 (Decimal('1997'), Decimal('2'), -37.17785290199861),
 (Decimal('1997'), Decimal('3'), 0.16522649038887202),
 (Decimal('1997'), Decimal('4'), 37.579187910257275),
 (Decimal('1997'), Decimal('5'), 1.4110800973551207),
 (Decimal('1997'), Decimal('6'), -32.38763433709323),
 (Decimal('1997'), Decimal('7'), 40.31057631048775),
 (Decimal('1997'), Decimal('8'), -7.316983704141531),
 (Decimal('1997'), Decimal('9'), 17.64005874784288),
 (Decimal('1997'), Decimal('10'), 19.98945679265288),
 (Decimal('1997'), Decimal('11'), -34.780054357730286),
 (Decimal('1997'), Decimal('12'), 64.00685004404939),
 (Decimal('1998'), Decimal('1'), 31.966644412344674),
 (Decimal('1998'), Decimal('2'), 5.511633272346428),
 (Decimal('1998'), Decimal('3'), 5.47085640480519),
 (Decimal('1998'), Decimal('4'), 18.06750267107856),
 (Decimal('1998'), Decimal('5'), -85.1907709370056)]

The results show the month-over-month sales growth rates for each month across multiple years. The growth rates are expressed as percentages, indicating the increase or decrease in sales from one month to the next.

Key insights from the data include:

- In some months, there is significant growth in sales, such as in October 1996 (42.21%) and December 1997 (64.01%), which are notably high growth periods.
- There are also months with substantial decreases in sales, like in February 1997 (-37.18%) and May 1998 (-85.19%), indicating periods of contraction.
- Overall, the data reveals fluctuating sales trends over time, with alternating periods of growth and decline.

This analysis of month-over-month sales growth rates provides valuable insights into sales performance trends, highlighting both strong and weak periods. Understanding these trends can assist in strategic decision-making and forecasting for future business operations.

###  Identifying High-Value Customers


To identify high-value customers for targeted promotions and special offers, I will analyse customer order values to identify those with above-average spending habits. These high-value customers are crucial for driving increased sales, improving customer retention, and attracting new customers to the business.

Using SQL queries and analytical techniques, I will calculate the average order value across all customers and then identify customers whose order values exceed this average. This segmentation will help prioritise marketing efforts towards these valuable customers, tailoring promotions and offers to their specific preferences and purchasing behaviour.

By focusing on high-value customers, the management team can maximise the impact of marketing initiatives, enhance customer satisfaction, and ultimately drive sustainable business growth.

In [48]:
%%sql

-- CTE to calculate total sales for each customer order
WITH customer_order_value AS (
    SELECT 
        o.customer_id, o.order_id,
        SUM(od.unit_price * od.quantity * (1 - od.discount)) AS total_sales
    FROM 
        orders AS o
    JOIN 
        order_details AS od ON o.order_id = od.order_id
    GROUP BY 
        o.customer_id, o.order_id
)

-- Selecting customer ID, total sales, and categorising based on sales compared to average
SELECT customer_id, total_sales,
    CASE
        WHEN total_sales > AVG(total_sales) OVER() THEN 'Above Average'
        ELSE 'Average/Below Average'
    END AS value_category
 FROM customer_order_value
ORDER BY total_sales DESC
LIMIT 10;


 * postgresql://postgres:***@localhost:5432/northwind
10 rows affected.


[('QUICK', 16387.49998714775, 'Above Average'),
 ('HANAR', 15810.0, 'Above Average'),
 ('SAVEA', 12615.050067901611, 'Above Average'),
 ('RATTC', 11380.0, 'Above Average'),
 ('SIMOB', 11188.400139808655, 'Above Average'),
 ('KOENE', 10952.84492739618, 'Above Average'),
 ('HUNGO', 10835.240051269531, 'Above Average'),
 ('RATTC', 10495.60012435913, 'Above Average'),
 ('QUICK', 10191.699981689453, 'Above Average'),
 ('QUICK', 10164.800018310547, 'Above Average')]

The analysis of customer order values has revealed several high-value customers who demonstrate above-average spending habits. These customers are prime candidates for targeted promotions and special offers, which can drive increased sales, enhance customer retention, and attract new customers to the business.

Some notable high-value customers based on their total sales and value category include:
- (QUICK) with total sales of 16387
- (HANAR) with total sales of 15810
- (SAVEA) with total sales of 12615

And several others listed in the "Above Average" value category.

These high-value customers represent significant revenue opportunities for the company, and targeted marketing strategies can be tailored to their preferences and purchasing behaviors to maximize their potential value to the business.

### Percentage of Sales for Each Category

Now I will be focusing on analysing product and category performance. By calculating the percentage of total sales for each category, I will be able to understand which categories contribute the most to overall sales. This analysis will be crucial for making informed decisions about inventory management, such as determining which categories require heavier stocking, and for devising targeted marketing strategies, such as identifying categories that should be promoted more aggressively.

In [66]:
%%sql

-- Creating a CTE to calculate total sales for each category
WITH total_sales_for_category AS (
    SELECT 
        c.category_id, c.category_name, 
        SUM(p.unit_price * od.quantity * (1 - od.discount)) AS total_sales
     FROM categories AS c
     JOIN products AS p ON c.category_id = p.category_id
     JOIN order_details AS od ON p.product_id = od.product_id
    GROUP BY c.category_id, c.category_name
)

-- Select category_name, total_sales, and calculate sales_percentage
SELECT category_name, total_sales, 
       (total_sales / SUM(total_sales) OVER ()) * 100 AS sales_percentage
  FROM total_sales_for_category 
 ORDER BY sales_percentage DESC;

 * postgresql://postgres:***@localhost:5432/northwind
8 rows affected.


[('Beverages', 288694.64971498586, 21.331025404054813),
 ('Dairy Products', 251147.5991296624, 18.556754766640605),
 ('Confections', 179863.64995904, 13.289737410548023),
 ('Meat/Poultry', 174622.5483462349, 12.902483709246834),
 ('Seafood', 137989.30574738482, 10.195732374296789),
 ('Condiments', 113692.18799885901, 8.400470714786334),
 ('Produce', 105745.70397348158, 7.813322138303922),
 ('Grains/Cereals', 101646.94242513739, 7.510473482122698)]

This analysis offers a comprehensive view of how sales are distributed among various product categories, enabling informed choices about inventory management and the development of focused marketing strategies.

### Top Products Per Category

I will be identifying the top three items sold in each product category. This will make it possible to identify start performers and ensure that these products are kept in stock and marketed prominently.

In [78]:
%%sql

-- Calculate total sales for each product in each category
WITH product_total_sales AS (
     SELECT c.category_id, c.category_name, p.product_id, p.product_name,
            SUM(p.unit_price * od.quantity * (1 - od.discount)) AS total_sales
       FROM categories AS c
       JOIN products AS p ON c.category_id = p.category_id
       JOIN order_details AS od ON p.product_id = od.product_id
      GROUP BY c.category_id, p.product_id, c.category_name, p.product_name
),

-- Rank products within each category based on total sales
ranked_products AS (
    SELECT category_name, product_name, total_sales,
           ROW_NUMBER() OVER(PARTITION BY category_name ORDER BY total_sales DESC) AS rank_of_product
      FROM product_total_sales
)

-- Select top 3 products within each category based on sales rank
SELECT category_name, product_name, total_sales
  FROM ranked_products
 WHERE rank_of_product <= 3;

 * postgresql://postgres:***@localhost:5432/northwind
24 rows affected.


[('Beverages', 'Côte de Blaye', 153897.1748863291),
 ('Beverages', 'Ipoh Coffee', 25109.09997367859),
 ('Beverages', 'Chang', 17719.399970583618),
 ('Condiments', 'Vegie-spread', 18343.61561246872),
 ('Condiments', "Sirop d'érable", 15022.349960759282),
 ('Condiments', 'Louisiana Fiery Hot Pepper Sauce', 14893.926944906489),
 ('Confections', 'Tarte au sucre', 50737.09416846588),
 ('Confections', "Sir Rodney's Marmalade", 24199.559986554086),
 ('Confections', 'Gumbär Gummibärchen', 21662.689146941742),
 ('Dairy Products', 'Raclette Courdavault', 76683.74989898875),
 ('Dairy Products', 'Camembert Pierrot', 49877.31995112449),
 ('Dairy Products', 'Mozzarella di Giovanni', 27086.57939014256),
 ('Grains/Cereals', 'Gnocchi di nonna Alice', 45351.09995948523),
 ('Grains/Cereals', 'Wimmers gute Semmelknödel', 23487.467487137765),
 ('Grains/Cereals', 'Singaporean Hokkien Fried Mee', 8986.599987879395),
 ('Meat/Poultry', 'Thüringer Rostbratwurst', 84783.77159642408),
 ('Meat/Poultry', 'Alice Mutton', 35105.849979020655),
 ('Meat/Poultry', 'Perth Pasties', 22623.799456167217),
 ('Produce', 'Manjimup Dried Apples', 43846.89994909987),
 ('Produce', 'Rössle Sauerkraut', 27936.839044377804),
 ('Produce', "Uncle Bob's Organic Dried Pears", 22453.49998757243),
 ('Seafood', 'Carnarvon Tigers', 30728.12496125698),
 ('Seafood', 'Ikura', 21653.499964892864),
 ('Seafood', 'Boston Crab Meat', 19055.039585784674)]

## Conclusion

My analysis of the Northwind Traders database has provided valuable insights into various aspects of the business operations. Key findings and conclusions from my analyses include:

1. Employee Sales Performance: I identified top-performing employees based on sales amount, highlighting individuals who significantly contribute to overall sales.

2. Sales Trends: Through analysing the running total of monthly sales and month-on-month sales growth rates, I observed consistent growth in sales over time, with notable seasonal variations.

3. High-Value Customers: By analysing customer order values, I identified high-value customers whose contributions to revenue are significant. Targeting and nurturing these customers can lead to increased customer retention and sales.

4. Product Sales Distribution: I analysed the percentage of sales for each product category, which revealed insights into sales distribution across different product types. This information can guide inventory management and marketing strategies.

5. Top Products: I identified top-selling products within each category, providing insights into customer preferences and popular product lines.

Based on these findings, I recommend the following actionable steps:

1. Employee Incentive Programmes: Consider implementing incentive programmes or recognition schemes for top-performing employees to motivate and reward sales excellence.

2. Customer Segmentation: Develop targeted marketing campaigns or personalised offerings for high-value customers to enhance customer satisfaction and loyalty.

3. Inventory Management: Use sales distribution insights to optimise inventory levels for different product categories, ensuring adequate stock of popular items while minimising excess inventory costs.

4. Product Focus: Allocate resources and marketing efforts towards promoting top-selling products to capitalise on customer preferences and market demand.

In conclusion, leveraging data-driven insights can drive strategic decision-making and improve overall business performance for Northwind Traders.